sdk/metric/controller/basic/config.go

```diff
@@ -68,11 +67,7 @@ type Option interface {
 // WithResource sets the Resource configuration option of a Config by merging it
 // with the Resource configuration in the environment.
 func WithResource(r *resource.Resource) Option {
-	res, err := resource.Merge(resource.Environment(), r)
```

Comment:
Why is this being removed?

Reply:
This was a mistake, I restored the existing code.

I consider these few lines to be bad behavior, but I didn't mean to change it in this PR -- it happened while I was revising the tests. There is already a:

```
	if c.Resource == nil {
		c.Resource = resource.Default()
	}
```

that will catch the environment and other built-in variables if none is provided when the controller is built. For a user's `WithResource()` to automatically add the environment seems to go against the specification [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#sdk-provided-resource-attributes).

I suspect what was _supposed_ to be done in this code was to merge resources if successive `WithResource()` calls are made. That change should happen in `apply()`. Since these tests are a lot of trouble to maintain, I applied this fix and updated the CHANGELOG.

Reply:
I think this is [required by the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable):

> The SDK MUST extract information from the OTEL_RESOURCE_ATTRIBUTES environment variable and merge this, as the secondary resource, with any resource information provided by the user, i.e. the user provided resource information has higher priority.

`resource.Default()` is only used when no user-provided resource exists, but when the user does provide one then the resource attributes detected from the environment must be merged in with it.

Reply:
OK. (Does this mean the user has no way of ignoring the environment?)
I will fix this.

Reply:
Fixed.

Reply:
Note: This support is moved into the `New()` function, with the same behavior as the original.